nexus/bin/qdens

```diff
@@ -554,9 +554,9 @@ class StatFile(QDBase):
         self.density_data = obj()
         for name in h5.keys():
             lname = name.lower()
-            if 'spin' in lname and 'density' in lname:
+            if lname == 'spindensity':
                 self.spin_density_data[name] = h5[name]
-            elif 'density' in lname:
```

Comment:
What is the value of `name` that the new code makes a difference with?

Reply:
Here are the `name` values I get:
```
ElecElec
IonIon
Kinetic
LocalECP
LocalEnergy
LocalEnergy_sq
LocalPotential
MPC
NEEnergyDensityEstimator
NonLocalECP
OneBodyDensityMatrices
SpinDensity
```
The relevant qmcpack input section is as follows:
```
      <estimators>
         <estimator type="spindensity" name="SpinDensity">
            <parameter name="grid">
               20 20 20
            </parameter>
         </estimator>
         <estimator type="EnergyDensity" name="EDcell" dynamic="e" static="ion0">
            <spacegrid coord="cartesian">
               <origin p1="zero"/>
               <axis p1="a1" scale=".5" label="x" grid="-1 (20) 1"/>
               <axis p1="a2" scale=".5" label="y" grid="-1 (20) 1"/>
               <axis p1="a3" scale=".5" label="z" grid="-1 (20) 1"/>
            </spacegrid>
         </estimator>
         <estimator type="OneBodyDensityMatrices" name="DensityMatrices">
            <parameter name="energy_matrix"       >    no                    </parameter>
            <parameter name="integrator"          >    density               </parameter>
            <parameter name="basis"               >
               spo_dm
            </parameter>
            <parameter name="evaluator"           >    matrix                </parameter>
            <parameter name="check_overlap"       >    no                    </parameter>
            <parameter name="samples"             >    10                    </parameter>
         </estimator>
      </estimators>
```

Reply:
When name='SpinDensity', lname='spindensity'. I don't see why the new code behaves differently from the old code. I'm looking for what this PR fixes/changes. So far, I feel there is no difference. Did I miss anything?

Reply:
In the older code if I have `name = OneBodyDensityMatrices`, then `'density' in lname` return True since the string 'density' is found in `onebodydensitymatrices`.  But qdens is not supposed to do anything with 1RDM and should skip this estimator. The new code explicitly checks if lname is equal to 'density'. So this newer one is a more stricter condition. I am assuming that the stat.h5 density estimator name will be exactly 'density'.

Reply:
Now I see.